src/platform/icelake/include/platform/dai.h

```diff
@@ -55,6 +55,11 @@
 /** \brief Number of HD/A Link Inputs */
 #define DAI_NUM_HDA_IN		7
 
+/* SoundWire */
+
+/** \brief Number of SoundWire multi-gateways */
+#define DAI_NUM_SOUNDWIRE_MULTI_GTW	16
```

Comment:
Also what is this value of 16 all about? Is this 4*4 for PCM?
You can have 6 PDIs per link (2 for bulk and 4 for PCM), plus 4 for PDM in. This should really be based on concepts such as BSS, ISS, OSS for PCM and PDM.

Reply:
@plbossart We have 4 segments and each has 4 bi-directional ports. Technically we have more ports available (there are also input only and output only ports), we're just ignoring them for now.